test/functional/feature_block.py

```diff
@@ -146,20 +146,6 @@ def run_test(self):
             badtx = template.get_tx()
             if TxTemplate != invalid_txs.InputMissing:
                 self.sign_tx(badtx, attempt_spend_tx)
-            else:
-                # Segwit is active in regtest at this point, so to deserialize a
-                # transaction without any inputs correctly, we set the outputs
-                # to an empty list. This is a hack, as the serialization of an
-                # empty list of outputs is deserialized as flags==0 and thus
-                # deserialization of the outputs is skipped.
```

Comment:
Please move this comment to `data/invalid_txs.py`, as it is not clear why the tx there is completely "empty"

Reply:
I wrote something else in its place. I found the previous text a little baffling to be honest. Willing to take heavy edits on it :)